Q24ECU/core/include/interfaces/interface_uart.h

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file interface_uart.h
+ * @author Jacob Chisholm (Jchisholm204.github.io)
+ * @brief uart FreeRTOS Interface with Semaphore Handler
+ * @version 0.1
+ * @date 2023-11-15
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+#include "hal/hal_uart.h"
+#include "FreeRTOS.h"
+#include "semphr.h"
+#include "stream_buffer.h"
+
+typedef struct {
+    USART_TypeDef *interface;
+    xSemaphoreHandle handle;
+    StreamBufferHandle_t stream;
+} uart_t;
+
+// handler for DEBUG USART (USART 2 on MockECU)
+extern uart_t debug;
+
+extern void uart_setup();
+
+// USART2 IQR Handler
+extern void USART2_IRQHandler();
+
+
+/**
+ * @brief Initialize a USART device and its Semaphore
+ * 
+ * @param handle The USART_t typedef handle
+ * @param interface the USARTx define from CMSIS headers
+ * @param baud USART Baud Rate
+ */
+static inline void uart_send_init(uart_t *handle, USART_TypeDef *interface, unsigned long baud){
+    handle->handle = xSemaphoreCreateBinary();
```

Comment:
having 'handle' as the semaphorehandle property and the parameter name is slightly confusing. Perhaps renaming one or both to be more specific would help